internal/routing/reader.go

```diff
@@ -88,6 +88,56 @@ func (r *routing) LocalSubnet() (defaultSubnet net.IPNet, err error) {
 	return defaultSubnet, fmt.Errorf("cannot find default subnet in %d routes", len(routes))
 }
 
+func (r *routing) LocalSubnets() (localSubnets []net.IPNet, err error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return localSubnets, fmt.Errorf("cannot find local subnet: %w", err)
+	}
+
+	localLinks := make(map[int]bool)
+
+	for _, link := range links {
+		if link.Attrs().EncapType != "ether" {
+			continue
+		}
+
+		localLinks[link.Attrs().Index] = true
+		if r.verbose {
+			r.logger.Info("local ethernet link found: %s", link.Attrs().Name)
+		}
+	}
+
+	if len(localLinks) == 0 {
+		return localSubnets, fmt.Errorf("cannot find any local interfaces")
+	}
+
+	routes, err := netlink.RouteList(nil, netlink.FAMILY_ALL)
+	if err != nil {
+		return localSubnets, fmt.Errorf("cannot list local routes: %w", err)
+	}
+
+	localSubnets = make([]net.IPNet, 1)
```

Comment:
That will blow up 😉

You are making localSubnets as `[]net.IPNet{net.IPNet{}}` here, and then append to that. Effectively you will end up with `[]net.IPNet{net.IPNet{}, yourActualIPNetYouCareAbout}`.

Just make it 0 sized so the append will work as intended (leave it the default return value which is already a zero sized slice).

```suggestion
```

You could also, if you really need memory performance, assign it a certain capacity with `localSubnets = make([]net.IPNet, 0, 10)` for example but that's not super clear and we really don't need it in this particular code section 😉

Reply:
Thanks! Yeah I found out about this and explicitly made it zero sized, but I didn't realise that it's already declared as part of the return value (fun times with go!), will remove it!